### PR TITLE
Update ApplicationContext Documentation

### PR DESCRIPTION
**Pull request by JustaSqu1d**
## Summary

Simply just added missing documentation for ApplicationContext.

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

File: discord/commands/context.py [hunk 1]
```diff
@@ -136,51 +137,67 @@ async def invoke(
...
         return self.interaction.channel
 
     @cached_property
     def channel_id(self) -> Optional[int]:
+        """`int`: Returns the ID of the channel associated with this context's command. Shorthand for :attr:`.Interaction.channel.id`."""
         return self.interaction.channel_id
 
     @cached_property
     def guild(self) -> Optional[Guild]:
+        """Optional[:class:`.Guild`]: Returns the guild associated with this context's command. Shorthand for :attr:`.Interaction.Guild`.
```
Comment (plun1331) [inline discord/commands/context.py:151]:
```suggestion
        """Optional[:class:`.Guild`]: Returns the guild associated with this context's command. Shorthand for :attr:`.Interaction.guild`.
```

File: discord/commands/context.py [hunk 2]
```diff
@@ -136,51 +137,67 @@ async def invoke(
 
     @cached_property
     def channel(self) -> Optional[InteractionChannel]:
+        """Union[:class:`.abc.Messageable`]: Returns the channel associated with this context's command. Shorthand for :attr:`.Interaction.channel`.
+        """
         return self.interaction.channel
 
     @cached_property
     def channel_id(self) -> Optional[int]:
+        """`int`: Returns the ID of the channel associated with this context's command. Shorthand for :attr:`.Interaction.channel.id`."""
```
Comment (plun1331) [inline discord/commands/context.py:146]: pretty sure this should also have :class:
```suggestion
        """:class:`int`: Returns the ID of the channel associated with this context's command. Shorthand for :attr:`.Interaction.channel.id`."""
```

File: discord/commands/context.py [hunk 3]
```diff
@@ -136,51 +137,67 @@ async def invoke(
...
 
     @cached_property
     def guild(self) -> Optional[Guild]:
+        """Optional[:class:`.Guild`]: Returns the guild associated with this context's command. Shorthand for :attr:`.Interaction.Guild`.
+        """
         return self.interaction.guild
 
     @cached_property
     def guild_id(self) -> Optional[int]:
+        """`int`: Returns the ID of the guild associated with this context's command. Shorthand for :attr:`.Interaction.Guild.id`."""
```
Comment (plun1331) [inline discord/commands/context.py:157]:
```suggestion
        """:class:`int`: Returns the ID of the guild associated with this context's command. Shorthand for :attr:`.Interaction.guild.id`."""
```

File: discord/commands/context.py [hunk 4]
```diff
@@ -136,51 +137,67 @@ async def invoke(
...
         return self.interaction.guild_id
 
     @cached_property
     def locale(self) -> Optional[str]:
+        """`str`: Returns the locale of the guild associated with this context's command. Shorthand for :attr:`.Interaction.locale`."""
         return self.interaction.locale
 
     @cached_property
     def guild_locale(self) -> Optional[str]:
+        """`str`: Returns the locale of the guild associated with this context's command. Shorthand for :attr:`.Interaction.Guild.locale`."""
```
Comment (plun1331) [inline discord/commands/context.py:167]:
```suggestion
        """:class:`str`: Returns the locale of the guild associated with this context's command. Shorthand for :attr:`.Interaction.guild_locale`."""
```

File: discord/commands/context.py [hunk 5]
```diff
@@ -136,51 +137,67 @@ async def invoke(
...
         return self.interaction.guild
 
     @cached_property
     def guild_id(self) -> Optional[int]:
+        """`int`: Returns the ID of the guild associated with this context's command. Shorthand for :attr:`.Interaction.Guild.id`."""
         return self.interaction.guild_id
 
     @cached_property
     def locale(self) -> Optional[str]:
+        """`str`: Returns the locale of the guild associated with this context's command. Shorthand for :attr:`.Interaction.locale`."""
```
Comment (plun1331) [inline discord/commands/context.py:162]:
```suggestion
        """:class:`str`: Returns the locale of the guild associated with this context's command. Shorthand for :attr:`.Interaction.locale`."""
```

File: discord/commands/context.py [hunk 6]
```diff
@@ -136,51 +137,67 @@ async def invoke(
...
     @cached_property
     def user(self) -> Optional[Union[Member, User]]:
+        """Union[:class:`.Member`, :class:`.User`]: Returns the user that sent this context's command. Shorthand for :attr:`.Interaction.user`."""
         return self.interaction.user
 
     author: Optional[Union[Member, User]] = user
 
     @property
     def voice_client(self) -> Optional[VoiceProtocol]:
+        """Optional[:class:`.VoiceProtocol`]: Returns the voice client associated with this context's command. Shorthand for :attr:`.Interaction.Guild.voice_client`, if applicable."""
```
Comment (plun1331) [inline discord/commands/context.py:192]:
```suggestion
        """Optional[:class:`.VoiceProtocol`]: Returns the voice client associated with this context's command. Shorthand for :attr:`.Interaction.guild.voice_client`, if applicable."""
```

**Review by plun1331 — CHANGES_REQUESTED** [review]

**Review by Lulalaby — DISMISSED** [review]

File: discord/commands/context.py [hunk 7]
```diff
@@ -136,51 +137,67 @@ async def invoke(
 
     @cached_property
     def channel(self) -> Optional[InteractionChannel]:
+        """Union[:class:`.abc.Messageable`]: Returns the channel associated with this context's command. Shorthand for :attr:`.Interaction.channel`.
+        """
```
Comment (Dorukyum) [inline discord/commands/context.py:141]:
```suggestion
        """Union[:class:`.abc.Messageable`]:
        Returns the channel associated with this context's command. Shorthand for :attr:`.Interaction.channel`."""
```

Comment (Dorukyum) [inline discord/commands/context.py:141]:
```suggestion
        """Union[:class:`abc.GuildChannel`, :class:`PartialMessageable`, :class:`Thread`]:
        Returns the channel associated with this context's command. Shorthand for :attr:`.Interaction.channel`."""
```

File: discord/commands/context.py [hunk 8]
```diff
@@ -136,51 +137,67 @@ async def invoke(
...
 
     @cached_property
     def channel_id(self) -> Optional[int]:
+        """:class:`int`: Returns the ID of the channel associated with this context's command. Shorthand for :attr:`.Interaction.channel.id`."""
         return self.interaction.channel_id
 
     @cached_property
     def guild(self) -> Optional[Guild]:
+        """Optional[:class:`.Guild`]: Returns the guild associated with this context's command. Shorthand for :attr:`.Interaction.guild`.
+        """
```
Comment (Dorukyum) [inline discord/commands/context.py:152]:
```suggestion
        """Optional[:class:`.Guild`]: Returns the guild associated with this context's command. Shorthand for :attr:`.Interaction.guild`."""
```

**Review by Dorukyum — CHANGES_REQUESTED** [review]

**Review by Dorukyum — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]
LGTM

**Pull request merged**